### PR TITLE
do not double uri encode

### PR DESCRIPTION
**Pull request by lperlaki**
the only places where `create_canonical_request`or `create_presigned_canonical_request` are used, pass in a uri_path wich comes from an `http::uri::Uri` and that already checks that the path is uri encoded.

### CODE REVIEW

File: crates/s3s/src/signature_v4/methods.rs [hunk 1]
```diff
@@ -137,7 +138,7 @@ pub fn create_canonical_request(
 
     {
         // <CanonicalURI>\n
-        uri_encode(&mut ans, uri_path, false);
+        ans.push_str(uri_path);
```
Comment (Nugine) [inline crates/s3s/src/signature_v4/methods.rs]: [AWS signature v4](https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html) defines their custom uri encoding function. The uri path from `hyper` may use different encoding. So the problem is not here.

We should decode the uri path and then pass it to the signature functions.

**Review by Nugine — CHANGES_REQUESTED** [review]
I suggest this fix:

1. Extract the decoded uri path
https://github.com/Nugine/s3s/blob/570edc821f2930403291188b9bf2efa1fe5a1d53/crates/s3s/src/ops/mod.rs#L45

2 . Put the decoded uri path into `SignatureContext`
https://github.com/Nugine/s3s/blob/570edc821f2930403291188b9bf2efa1fe5a1d53/crates/s3s/src/ops/mod.rs#L252-L262

3. Change the corresponding usages
https://github.com/Nugine/s3s/blob/570edc821f2930403291188b9bf2efa1fe5a1d53/crates/s3s/src/ops/mod.rs#L351

**Comment by Nugine** [comment]
It's quite easy to forget whether the uri from `hyper` is decoded.
+ https://github.com/hyperium/http/issues/384

**Comment by lperlaki** [comment]
ok implement it like you suggested.

**Review by Nugine — APPROVED** [review]
Thanks!

**Pull request merged**